loadtest/loadtest.go

```diff
@@ -0,0 +1,328 @@
+
+		err = l.installTestApp(ctx, loadTestEndpoint)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", "installed loadtest app")
+	}
+
+	{
+		l.logger.LogCtx(ctx, "level", "debug", "message", "waiting for loadtest app to be ready")
+
+		err = l.waitForLoadTestApp(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", "loadtest app is ready")
+	}
+
+	return nil
+}
+
+// enableIngressControllerHPA enables HPA via the user configmap and updates
+// the chartconfig CR so chart-operator reconciles the config change.
+func (l *LoadTest) enableIngressControllerHPA(ctx context.Context) error {
+	var err error
+
+	l.logger.Log("level", "debug", "message", fmt.Sprintf("waiting for %#q configmap to be created", UserConfigMapName))
+
+	o := func() error {
+		lo := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app=%s,cluster-operator.giantswarm.io/configmap-type=user", UserConfigMapName),
+		}
+		l, err := l.guestFramework.K8sClient().CoreV1().ConfigMaps(metav1.NamespaceDefault).List(lo)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(l.Items) != 1 {
+			return microerror.Maskf(waitError, "want %d configmaps found %d", 1, len(l.Items))
+		}
+
+		return nil
+	}
+
+	b := backoff.NewConstant(10*time.Minute, 15*time.Second)
+	n := func(err error, delay time.Duration) {
+		l.logger.Log("level", "debug", "message", err.Error())
+	}
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	l.logger.Log("level", "debug", "message", fmt.Sprintf("waited for %#q configmap to be created", UserConfigMapName))
+
+	values := UserConfigMapValues{
+		Data: UserConfigMapValuesData{
+			AutoscalingEnabled: true,
+		},
+	}
+
+	var data []byte
+
+	data, err = yaml.Marshal(values)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = l.guestFramework.K8sClient().CoreV1().ConfigMaps(metav1.NamespaceSystem).Patch(UserConfigMapName, types.StrategicMergePatchType, data)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var cr *v1alpha1.ChartConfig
+
+	cr, err = l.guestFramework.G8sClient().CoreV1alpha1().ChartConfigs(CustomResourceNamespace).Get(CustomResourceName, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Set dummy annotation to trigger an update event.
+	annotations := cr.Annotations
+	annotations["test"] = "test"
+	cr.SetAnnotations(annotations)
+
+	_, err = l.guestFramework.G8sClient().CoreV1alpha1().ChartConfigs(CustomResourceNamespace).Update(cr)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// installLoadTestApp installs a chart that deploys the Stormforger test app
+// in the tenant cluster as the test workload for the load test.
+func (l *LoadTest) installTestApp(ctx context.Context, loadTestEndpoint string) error {
+	var err error
+
+	var jsonValues []byte
+	{
+		values := AppValues{
```

Comment:
Nit: I would use `map[string]interface{}` here to reduce some clutter as the type isn't reused anyway.

Reply:
That's nice, thanks. There are a few of these structs currently.